### PR TITLE
consolidate Go SDK activities content in one doc

### PR DESCRIPTION
**Pull request by sw-yx**
## DO NOT MERGE 

These changes are also captured here: https://github.com/temporalio/documentation/pull/340
Keep this open for tracking until https://github.com/temporalio/documentation/pull/340 is merged.

## What was changed:

- moved content from 2 docs into `go-activities`.
- slight rewrite to reduce "wall of text"

> NOTE: we want to use snipsync to replace the sample code, i havent done it yet, this PR is just consolidating existing content

See new Go SDK activities doc: https://deploy-preview-330--mystifying-fermi-1bc096.netlify.app/docs/go-activities

## Why?

previously our activities docs were all over the place (legacy holdover from Cadence)

image

one doc should be the definitive source of and pointer to all the things i need to know on that topic, i shouldnt have to read thru all the things in the sidebar

### CODE REVIEW

File: docs/go-activities.md [hunk 1]
```diff
@@ -4,21 +4,7 @@ title: Activities
...
-value. The result value can be either a basic type or a struct with the only requirement being that
-it is serializable.
-
-The values passed to Activities through invocation parameters or returned through the result value
-are recorded in the execution history. The entire execution history is transferred from the Temporal
-service to Workflow workers with every event that the Workflow logic needs to process. A large execution
-history can thus adversely impact the performance of your Workflow. Therefore, be mindful of the amount
-of data you transfer via Activity invocation parameters or return values. Otherwise, no additional
-limitations exist on Activity implementations.
+You should have a [conceptual understanding of Temporal Activities](https://docs.temporal.io/docs/concept-activities/) before proceeding.
```
Comment (sw-yx) [inline docs/go-activities.md:7]: moved this down to avoid super lengthy prelude

File: docs/go-activities.md [hunk 2]
```diff
@@ -57,9 +43,19 @@ The Activity declares two return values: string and error. The string return val
...
+    - Data can be passed directly to an Activity via function
+parameters. The parameters can be either basic types or structs (must be serializable). 
+    - Though it is not required, we recommend that the first parameter
+of an Activity function is of type `context.Context`, in order to allow the Activity to interact with other framework methods. 
+- Return values:
+    - The function must return an `error` value.
+    - The result value is optional, and can be either a basic type or a struct (must be serializable).
+
+There's no hard limit on what you can pass into or return from an Activity function, but keep in mind that all parameters and return values are recorded in the execution history.
+A large execution history can adversely impact the performance of your Workflows as the entire history is transferred to your workers with every event processed.
```
Comment (sw-yx) [inline docs/go-activities.md:59]:
```suggestion
No other limitations on Activity functions exist; you are free to use idiomatic loggers and metrics controllers, and the standard Go concurrency constructs.
```

File: docs/go-activities.md [hunk 3]
```diff
@@ -126,6 +123,166 @@ w.registerActivity(ActivityB)
 w.registerActivity(ActivityC)
 ```
 
-## Failing an Activity
 
-To mark an Activity as failed, the Activity function must return an error via the `error` return value.
+## Synchronous Activity Execution
```
Comment (sw-yx) [inline docs/go-activities.md:127]: crappy orphan section. moved up into the return values bullet points

File: netlify.toml [hunk 4]
```diff
@@ -13,6 +13,14 @@
...
   to = "/docs/concept-activities"
 
+[[redirects]]
+  from = "/docs/go-execute-activity"
+  to = "/docs/go-activities#executing-activities"
+
+[[redirects]]
+  from = "/docs/go-activity-async-completion"
+  to = "/docs/go-activities#asynchronous-activity-completion"
+
```
Comment (sw-yx) [inline netlify.toml:23]: not sure it redirects to #'s properly, need to test

File: netlify.toml [hunk 5]
```diff
@@ -9,10 +9,22 @@
   from = "/docs/overview"
   to = "/docs/concept-overview"
 
+[[redirects]]
+  from = "/docs"
+  to = "/docs/get-started"
```
Comment (sw-yx) [inline netlify.toml:14]: image

**Comment by flossypurse** [comment]
Capturing these changes in the AprilSprint PR: https://github.com/temporalio/documentation/pull/340

**Comment by sw-yx** [comment]
didnt see the "keep this open" haha

**Comment by flossypurse** [comment]
We can close this now - incorporated changes in `aprilsprint` branch

**Pull request closed without merging**